AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,74 @@
 //
 
 import UIKit
+import AVFoundation
+import AVKit
+import MobileCoreServices
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
     var titleForCell: String = "Double Video"
-
+    var movieURL: URL?
+    var player: AVPlayer?
+    
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
+    
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
+       
     }
     
-    @IBAction func loadVideo(_ sender: UIButton) {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        
+        //guard let type = info[UIImagePickerControllerMediaType] as? String else { return }
+        //dump("TYPE>> \(type)")
+        
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeImage):
+            print("Image!")
+        case String(kUTTypeMovie):
+            print("Movie!")
+            if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+                self.movieURL = url
+            }
+        default:
+            print("bad media")
+        }
+        
+        // dismissing imagePickerController
+        dismiss(animated: true ) {
+            if let url = self.movieURL {
+                let playerItem = AVPlayerItem(url: url)
+                self.player = AVPlayer(playerItem: playerItem)
+                let playerLayer = AVPlayerLayer(player: self.player)
+                if self.videoContainerTop != nil {
```

Comment:
The containers are set and are never nil. Checking for the rate of the video played within is what would help us decide whether to use that container.